### PR TITLE
Introduce new annotation doc comment format

### PR DESCRIPTION
**Pull request by mogottsch**
closes https://github.com/giantswarm/crd-docs-generator/issues/124

## What does this PR do?
Because `go-fmt` breaks our current formatting of yaml embedded in doc comments, we adapt the format of annotation doc comments.

Before:
```
// support:
//   - crd: awsclusters.infrastructure.giantswarm.io
//     apiversion: v1alpha2
//     release: Since 14.0.0
// documentation:
//   This annotation allows configuration of the MINIMUM_IP_TARGET parameter for AWS CNI.
```
after:
```
// Comment above is possible
// CRD_DOCS_GENERATOR:
//
//	support:
//	  - crd: awsclusters.infrastructure.giantswarm.io
//	    apiversion: v1alpha2
//	    release: Since 14.0.0
//	documentation:
//	  This annotation allows configuration of the MINIMUM_IP_TARGET parameter for AWS CNI.
```

## Why this formatting?
Indentation in go doc comments is used for [code blocks](https://tip.golang.org/doc/comment). Therefore, we need to put our yaml code into a code block by indenting it by one tab.
This does not work unless there is some 'normal' text before the code block. The token `CRD_DOCS_GENERATOR` is this 'normal' text and simultaneously shows where the yaml code starts.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by marians** [comment]
If it works, it looks pretty good to me! I wonder whether we need to pay close attention to have a tab as the first indentation and then have spaces. That looks like dealing with white space in comments could become pretty brittle. Anyway, it's the best I see so far.

Will this also work if there is a blank line above `CRD_DOCS_GENERATOR`? Aesthetically I would prefer that.

**Comment by mogottsch** [comment]
> If it works, it looks pretty good to me! I wonder whether we need to pay close attention to have a tab as the first indentation and then have spaces. That looks like dealing with white space in comments could become pretty brittle. Anyway, it's the best I see so far.
> 
> Will this also work if there is a blank line above `CRD_DOCS_GENERATOR`? Aesthetically I would prefer that.

Go fmt indents all lines in a code block by one tab, so we will have to stick with that. We could also use tab inside our yaml, however, that would not look good I think. 

We can only insert a blank line above `CRD_DOCS_GENRATOR` if there is another non-emtpy comment line above that, because comments cannot start or end with an empty line.

So this is possible:
```
// Comment above is possible
//
// CRD_DOCS_GENERATOR:
//
//	support:
//	  - crd: awsclusters.infrastructure.giantswarm.io
//	    apiversion: v1alpha2
//	    release: Since 14.0.0
//	documentation:
//	  This annotation allows configuration of the MINIMUM_IP_TARGET parameter for AWS CNI.
```
But this is not:
```
//
// CRD_DOCS_GENERATOR:
//
//	support:
//	  - crd: awsclusters.infrastructure.giantswarm.io
//	    apiversion: v1alpha2
//	    release: Since 14.0.0
//	documentation:
//	  This annotation allows configuration of the MINIMUM_IP_TARGET parameter for AWS CNI.
```

**Comment by marians** [comment]
> We can only insert a blank line above `CRD_DOCS_GENRATOR` if there is another non-emtpy comment line above that, because comments cannot start or end with an empty line.

That's fine!

**Comment by marians** [comment]
This is a breaking change, right? No problem with that, but we should make sure to release it as a new major then.

**Review by marians — APPROVED** [review]

**Comment by mogottsch** [comment]
I talked to Vaclav about backward compatibility. As these are breaking changes, repositories with the old format will break.
I think this is ok because as soon as we update to the newest go version, those comments will break anyway.
What do you think about that @marians ?

File: pkg/annotations/annotations.go [hunk 1]
```diff
@@ -85,6 +87,49 @@ func Collect(startPath string) ([]CRDAnnotationSupport, error) {
...
+// removes leading tabs from each line
+func unIndent(lines []string) []string {
+	var result []string
+	for _, line := range lines {
+		if strings.HasPrefix(line, "\t") {
+			result = append(result, line[1:])
+		}
+	}
+
+	return result
```
[the remark this answers is not in the document]
Comment (mogottsch) [reply pkg/annotations/annotations.go:130]: Personally, I would prefer display and error message in that case.

[the remark this answers is not in the document]
Comment (mogottsch) [reply pkg/annotations/annotations.go:130]: Ok, nevermind. I realized that not all of the considered lines are indented (e.g. the first and last are empty). Therefore, I decided to always return the lines in the `unIndent` method. If there was malformed input, then there would be an error output, when parsing trying to parse the yaml.
Does that sound good to you?

[the remark this answers is not in the document]
Comment (mogottsch) [reply pkg/annotations/annotations.go:130]: Oops, forgot to push :sweat_smile:  Now you can look at it @vvondruska

[the remark this answers is not in the document]
Comment (vvondruska) [reply pkg/annotations/annotations.go:129]: Looks good to me

**Review by vvondruska — APPROVED** [review]

**Comment by mogottsch** [comment]
> This is a breaking change, right? No problem with that, but we should make sure to release it as a new major then.

Marian and I decided not to release a major version as the current version is still pre-release `0.X.X` [(semver reference)](https://semver.org/#spec-item-4).

**Pull request merged**